src/main/java/org/apache/commons/io/IOUtils.java

```diff
@@ -708,31 +706,79 @@ public static void closeQuietly(final Writer output) {
     @SuppressWarnings("resource")
     public static boolean contentEquals(final InputStream input1, final InputStream input2)
             throws IOException {
+        return contentEquals(input1, input2, DEFAULT_BUFFER_SIZE);
+    }
+
+    /**
+     * Compares the contents of two Streams to determine if they are equal or not.
+     * <p>
+     * This method buffers the input internally.
+     *
+     * @param input1 the first stream
+     * @param input2 the second stream
+     * @param bufferSize the size of the internal buffer to use.
+     * @return true if the content of the streams are equal or they both don't
+     * exist, false otherwise
+     * @throws NullPointerException if either input is null
```

Comment:
I fail to see where this one is thrown.

Reply:
This documentation is a clone/mutate from the original method's javadoc (the one without the bufferSize).  This doesn't look to be explicitly thrown, but if either InputStream is null it throws due to the stream being dereferenced.

Check the javadoc for the `public static boolean contentEquals(final InputStream input1, final InputStream input2)` method.

Happy to either (a) explicitly throw a NullPointerException up front, (b) remove the comment from both places, or (c) do nothing.

Reply:
I would rather see an explicit NPE up front.

Reply:
I looked at the beginning of each method again.  Both contentEquals methods start with the following statements.  This can never throw a NullPointerException.  If both streams are null they will be equal and therefore always return true.  If one stream is null and the other is not it will always return false based on the XOR logic.  Only if both streams are non-null and not the same object will the actual logic of the method execute.

```
if (input1 == input2) {
            return true;
}
if (input1 == null ^ input2 == null) {
            return false;
}
```